android/src/main/cpp/NativeProxy.cpp

```diff
@@ -530,8 +523,7 @@ void NativeProxy::setupLayoutAnimations() {
               yogaValues.setProperty(rt, key, value);
             }
           } catch (std::invalid_argument e) {
-            errorHandler->setError("Failed to convert value to number");
-            errorHandler->raise();
+            throw std::runtime_error("Failed to convert value to number");
```

Comment:
Did you tried to trigger this exception in application? I'm wondering if it crash app or just display RedBox with message.

Reply:
Turns out it's actually shown using RedBox! 😄 

<img width="489" alt="Zrzut ekranu 2023-07-12 o 22 29 43" src="https://github.com/software-mansion/react-native-reanimated/assets/20516055/33e4bc0a-a852-47da-8ab4-69aa91b923fb">